stores/migrations/mysql/main/migration_00002_prune_slabs_trigger.sql

```diff
@@ -0,0 +1,52 @@
+-- prune manually before creating trigger
+DELETE slabs
+FROM slabs
+LEFT JOIN slices ON slices.db_slab_id = slabs.id
+WHERE slices.db_object_id IS NULL
+AND slices.db_multipart_part_id IS NULL
+AND slabs.db_buffered_slab_id IS NULL;
+
+-- remove ON DELETE CASCADE from slices
+ALTER TABLE slices DROP FOREIGN KEY fk_objects_slabs;
+ALTER TABLE slices ADD CONSTRAINT fk_objects_slabs FOREIGN KEY (db_object_id) REFERENCES objects(id);
+
+ALTER TABLE slices DROP FOREIGN KEY fk_multipart_parts_slabs;
+ALTER TABLE slices ADD CONSTRAINT fk_multipart_parts_slabs FOREIGN KEY (db_multipart_part_id) REFERENCES multipart_parts(id);
+
+-- remove ON DELETE CASCADE from multipart_parts
+ALTER TABLE multipart_parts DROP FOREIGN KEY fk_multipart_uploads_parts;
+ALTER TABLE multipart_parts ADD CONSTRAINT fk_multipart_uploads_parts FOREIGN KEY (db_multipart_upload_id) REFERENCES multipart_uploads(id);
+
+-- dbObject trigger to delete from slices
+CREATE TRIGGER delete_from_slices_after_objects_delete
```

Comment:
`nit` `after` is misleading here, I'd rename all these triggers to `[table]_on_[event]_[action]` so it would become `objects_on_before_delete_delete_slices` (or maybe `..._remove_slices`)

Reply:
oh yeah that's a leftover from when it was a AFTER trigger. Changed the naming to something similar to what you suggested.